library/pod-security-policy/forbidden-sysctls/src.rego

```diff
@@ -0,0 +1,26 @@
+package k8spspforbiddensysctls
+
+violation[{"msg": msg, "details": {}}] {
+    sysctls := {x | x = input.review.object.spec.securityContext.sysctls[_][name]}
+    input_sysctls(sysctls)
+    msg := sprintf("One of the sysctls %v is not allowed, pod: %v. Forbidden sysctls: %v", [sysctls, input.review.object.metadata.name, input.parameters.forbiddenSysctls])
+}
+
+# * may be used to forbid all sysctls
+input_sysctls(sysctls) {
+    input.parameters.forbiddenSysctls[_] == "*"
```

Comment:
Should this check for non-zero length on sysctls?

Reply:
if it's non-zero, it either has `*` or list of systctls. If it's zero, we are allowing all sysctls. Is there a use case that I am missing that should check for non-zero?

Reply:
If an empty list means "allow all" this looks correct.

It's a little weird that empty means "allow all" as an admin could just avoid implementing the constraint altogether. I'm guessing this is legacy from translating directly from PSP, where this was one field of many and they wanted undefined to default to permissive?

It would be more natural to diverge from the PSP spec here and say: "empty list means deny all", I'm curious if @ritazh  has an opinion.

Reply:
> It's a little weird that empty means "allow all" as an admin could just avoid implementing the constraint altogether.

I agree it is a little counter intuitive, but this is currently consistently implemented across all the PSP CTs and constraints: 

https://github.com/open-policy-agent/gatekeeper/blob/master/library/pod-security-policy/flexvolume-drivers/src.rego#L10-L11 

as it is mirroring the expected behavior from PSP: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#flexvolume-drivers
> An empty list or nil means there is no restriction on the drivers

My personal opinion is we should map to the same documented default behavior as PSP, otherwise it could get very confusing what the default behavior should be. WDYT?

Reply:
SGTM